addons/packages/harbor/2.4.2/test/e2e/harbor_suite_test.go

```diff
@@ -0,0 +1,425 @@
+	DisplayName string // the display name of the package, used to find the package name
+	ValuesFile  string // the custom values file when installing the package
+}
+
+const (
+	packagePollInterval = "10s"
+	packagePollTimeout  = "20m"
+)
+
+var (
+	// packageInstallNamespace is the namespace where the harbor package
+	// is installed (i.e this is the namespace tanzu package install is called
+	// with). Optionally provided by using PACKAGE_INSTALL_NAMESPACE env var.
+	// If PACKAGE_INSTALL_NAMESPACE is not provided the test assumes the available
+	// package is in the global namespace and the test will create and install
+	// the package into its own test namespace.
+	// If PACKAGE_INSTALL_NAMESPACE is provided the test assumes
+	// the available package is namespaced and will install the package into
+	// the provided namespace and will NOT delete that namespace.
+	// Read more on package namespacing here:
+	// https://carvel.dev/kapp-controller/docs/latest/package-consumer-concepts/#overview
+	packageInstallNamespace string
+
+	// packageComponentsNamespace is the namespace where the harbor
+	// package components are installed  (e.g. the  deployment).
+	// This is the namespace passed into the harbor values.yaml). This
+	// namespace is created by the package installation.
+	packageComponentsNamespace string
+
+	// packageInstallName is the app name of the harbor installed
+	packageInstallName string
+
+	// packageDependencies is the dependencies for the harbor package
+	packageDependencies []*packageDependency
+
+	// harborHostname the hostname of the harbor installed
+	harborHostname string
+
+	// harborAdminPassword the password of the harbor installed
+	harborAdminPassword string
+
+	// installedPackages record the packages installed by the test
+	installedPackages []string
+
+	// storage class name for PVC usage
+	storageclass string
+	isDefault    bool
+)
+
+var _ = BeforeSuite(func() {
+	packageInstallNamespace = "default"
+
+	packageInstallName = "harbor"
+
+	packageComponentsNamespace = "goharbor"
+
+	//storageclass = getStorageClass()
+	storageclass, isDefault = getAvailableStorageClass()
+	if isDefault {
+		storageclass = ""
+	}
+	fmt.Println("storageclass", storageclass)
+
+	packageDependencies = []*packageDependency{
+		{"cert-manager", "cert-manager", ""},
+		{"contour", "contour", configContourYamlFile()},
+	}
+
+	for _, dependency := range packageDependencies {
+		By(fmt.Sprintf("installing %s addon package", dependency.Name))
+
+		packageName := utils.TanzuPackageName(dependency.DisplayName)
+
+		version := findPackageAvailableVersion(packageName, "")
+		installPackage(dependency.Name, packageName, version, dependency.ValuesFile)
+	}
+
+	By("installing harbor addon package")
+	harborHostname = fmt.Sprintf("harbor.%s.nip.io", getReachableContourEnvoyIP())
+
+	harborAdminPassword = generateHarborPassword()
+
+	packageName := utils.TanzuPackageName("harbor")
+
+	version := findPackageAvailableVersion(packageName, "2.3.3")
```

Comment:
```suggestion
	version := findPackageAvailableVersion(packageName, "2.4.2")
```

I assume that this should be `2.4.2` for the `2.4.2` version of the package?